src/tools/rbd/action/DiskUsage.cc

```diff
@@ -217,19 +209,20 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
 
     for (std::vector<librbd::snap_info_t>::const_iterator snap =
          snap_list.begin(); snap != snap_list.end(); ++snap) {
-      librbd::Image snap_image;
+      librbd::Image snap_image;     
       r = rbd.open_read_only(io_ctx, snap_image, image_spec.name.c_str(),
```

Comment:
Now that we open an image rather than a snapshot here, it might be possible to move `snap_image` variable and this call outside of the loop so that the "snap" image is opened just once, independent of the number of snapshots.  It would be a nice optimization ;)